### PR TITLE
Contributing and issue reporting guidelines / templates

### PR DESCRIPTION
**Pull request by Tehem**
### Purpose of this PR

- Add IntelliJ and auto-generated documentation to .gitignore
- Add contributing guidelines (largely inspired by [Node.js](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md))
- Add issue reporting template (largely inspired by [Spotify](https://github.com/spotify) (I don't remember which repository in particular)

### Configuration change

None

### Checks

- [ ] Documentation of environment variables is very clear, even for a newcomer
- [ ] All relevant usecases of are tested
- [ ] These changes won't cause memory / CPU problems even if business grows by a factor of 100
- [ ] It's ok if these changes go to production and then are rolled-back

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+# Contributing to BusyBee
+
+This section will guide you through the contribution process.
+
+### Step 1: Fork
+
+Fork the project [on GitHub](https://github.com/ChauffeurPrive/busybee) and clone your fork
+locally.
+
+```text
```
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]:
```bash
 git clone git@github.com:username/busybee.git
 cd busybee
 git remote add upstream https://github.com/ChauffeurPrive/busybee.git
 ```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+For developing new features and bug fixes, the `master` branch should be pulled
+and built upon.
+
+### Step 2: Branch
+
+Create a branch and start hacking:
+
+```text
+$ git checkout -b my-branch -t origin/master
```
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]: `my-amazing-feature` :) ?

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+### Step 3: Commit
+
+Make sure git knows your name and email address:
+
+```text
+$ git config --global user.name "J. Random User"
+$ git config --global user.email "j.random.user@example.com"
+```
+
+Add and commit:
```
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]: typo ?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+```text
+$ git add my/changed/files
+$ git commit
+```
+
+### Commit message guidelines
+
+The commit message should describe what changed and why.
```
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]: Example ?

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+repository.
+
+```text
+$ git fetch upstream
+$ git rebase upstream/master
+```
+
+### Step 5: Test
+
+Bug fixes and features should come with tests. We are using the [mocha javascript
```
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]: should --> **must**

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+to delete history in git. Before you use it, make sure you understand the risks.
+
+Feel free to post a comment in the Pull Request to ping reviewers if you are
+awaiting an answer on something.
+
+### Step 8: Landing
+
+In order to land, a Pull Request needs to be reviewed and
+[approved](#getting-approvals-for-your-pull-request) by
+at least one Chauffeur Privé Collaborator and pass a
```
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]: Collaborator ? Not Maintainer ?

File: README.md [hunk 7]
```diff
@@ -1,6 +1,7 @@
 # BusyBee
 [![CircleCI](https://circleci.com/gh/ChauffeurPrive/busybee/tree/master.svg?style=shield&circle-token=9a4d0d25bd8e0134d33f386a66c90c80dd401cf1)](https://circleci.com/gh/ChauffeurPrive/busybee/tree/master)
 [![codecov](https://codecov.io/gh/ChauffeurPrive/busybee/branch/master/graph/badge.svg)](https://codecov.io/gh/ChauffeurPrive/busybee)
+[![License](https://img.shields.io/github/license/ChauffeurPrive/busybee.svg)](LICENSE)
```
Comment (GillesRasigade) [inline README.md]: +1

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (Tehem) [inline .github/CONTRIBUTING.md]: I don't see it...

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (Tehem) [inline .github/CONTRIBUTING.md]: replace text with bash ? is that what you mean ?

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (Tehem) [inline .github/CONTRIBUTING.md]: ok

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (GillesRasigade) [inline .github/CONTRIBUTING.md]: My bad sorry :(

**Pull request merged**